pkg/client/fibre_channel_configs_object_set.go

```diff
@@ -80,3 +80,48 @@ func buildFibreChannelConfigObjectSet(response interface{}) []*nimbleos.FibreCha
 
 	return results
 }
+
+// List of supported actions on object sets
+
+// Regenerate - Regenerate Fibre Channel configuration.
+func (objectSet *FibreChannelConfigObjectSet) RegenerateObjectSet(id *string, wwnnBaseStr *string, precheck *bool) (*nimbleos.NsFcConfigRegenerateReturn, error) {
+
+	regenerateUri := fibreChannelConfigPath
+	regenerateUri = regenerateUri + "/" + *id
+	regenerateUri = regenerateUri + "/actions/" + "regenerate"
+
+	payload := &struct {
+		Id          *string `json:"id,omitempty"`
+		WwnnBaseStr *string `json:"wwnn_base_str,omitempty"`
+		Precheck    *bool   `json:"precheck,omitempty"`
+	}{
+		id,
+		wwnnBaseStr,
+		precheck,
+	}
+
+	resp, err := objectSet.Client.Post(regenerateUri, payload, &nimbleos.NsFcConfigRegenerateReturn{})
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.(*nimbleos.NsFcConfigRegenerateReturn), err
+}
+
+// HwUpgrade - Update Fibre Channel configuration after hardware changes.
+func (objectSet *FibreChannelConfigObjectSet) HwUpgradeObjectSet(id *string) error {
+
+	hwUpgradeUri := fibreChannelConfigPath
+	hwUpgradeUri = hwUpgradeUri + "/" + *id
+	hwUpgradeUri = hwUpgradeUri + "/actions/" + "hw_upgrade"
+
+	payload := &struct {
+		Id *string `json:"id,omitempty"`
```

Comment:
See.  Here's another one.

Why can't this just be:

```
payload := &struct {
		Id *string 
                } {
                               id,
                }
```

Reply:
Oh.  It's probably because it won't serialize to snake_case over the wire right?

Reply:
@rgcostea , The Actions are not same on all the object set also there may be few tags that may be common but overall the structures are not common. Thus I made it local to specific action implementation since it cannot be generic.